bootstrap/v2/pkg/apis/apps/kfdef/v1alpha1/application_types.go

```diff
@@ -655,3 +681,101 @@ func IsSecretNotFound(e error) bool {
 	_, ok := e.(*SecretNotFound)
 	return ok
 }
+
+func getParameter(parameters []config.NameValue, paramName string) (string, bool) {
+	for _, p := range parameters {
+		if p.Name == paramName {
+			return p.Value, true
+		}
+	}
+
+	return "", false
+}
+
+func setParameter(parameters []config.NameValue, paramName string, value string) []config.NameValue {
+	pIndex := -1
+
+	for i, p := range parameters {
+		if p.Name == paramName {
+			pIndex = i
+		}
+	}
+
+	if pIndex < 0 {
+		parameters = append(parameters, config.NameValue{})
+		pIndex = len(parameters) - 1
+	}
+
+	parameters[pIndex].Name = paramName
+	parameters[pIndex].Value = value
+
+	return parameters
+}
+
+type AppNotFound struct {
```

Comment:
Should we declare a new type or reuse the KFError struct with a new error code ?

Reply:
I think it makes more sense to define a new type(s) here; I think that's more consistent with Go style. These errors are specific to this package.

I think if we want to define common error codes then we should consider making KFError an interface and then these errors can implement that interface.